apps/workflowengine/src/components/Checks/RequestUserGroup.vue

```diff
@@ -107,10 +120,32 @@ export default {
 				this.groups.push(group)
 			}
 		},
+		hasGroup(group) {
+			const index = this.groups.findIndex((item) => item.id === group)
+			return index > -1
+		},
 		update(value) {
 			this.newValue = value.id
 			this.$emit('update:model-value', this.newValue)
 		},
+		enqueueWantedGroup(expectedGroupId) {
+			const index = this.wantedGroups.findIndex((groupId) => groupId === expectedGroupId)
+			if (index === -1) {
+				this.wantedGroups.push(expectedGroupId)
+			}
+		},
+		async findGroupByQueue() {
+			let nextQuery
+			do {
+				nextQuery = this.wantedGroups.shift()
+				if (this.hasGroup(nextQuery)) {
+					nextQuery = undefined
+				}
+			} while (!nextQuery && this.wantedGroups.length > 0)
+			if (nextQuery) {
+				await this.searchAsync(nextQuery)
+			}
```

Comment:
Easier to read.

```suggestion
			while (this.wantedGroups.length > 0) {
				const groupId = this.wantedGroups.shift()
				if (this.hasGroup(groupId)) {
					continue
				}
				await this.searchAsync(groupId)
				return
			}
```

Also, the current implementation would only search 1 wanted group, I assuming this intended as `this.findGroupByQueue()` again called at the end of search

Reply:
This was intentionally done as a queue in order to not DoS the server with group search requests, but have them coming sequentially.